### PR TITLE
batch system -> job runner

### LINKED ISSUE #3910: Replace `batch system` with `job runner`
Spun out of issue #3696 .

As per the [config change proposal](https://cylc.github.io/cylc-admin/proposal-config-changes.html#job-platformsbatch-system---platformsjob-runner) rename the following config items:
- [ ] [job platforms][]batch system -> [platforms][]job runner
- [ ] [job platforms][]batch submit command template -> [platforms][]job runner command template

These items do not need upgraders.

Question: Cylc has a whole infrastructure of `batch system` stuff, so this could be a rabbit hole - how far down should we pursue this change now? @oliver-sanders - can I have your input on that aspect of it?

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by MetRonnie**
These changes close #3910. Yikes.

Global config change:
- `[platforms][X]batch system` -> `[platforms][X]job runner`
- `[platforms][X]batch submit command template` -> `[platforms][X]job runner command template`

**Plus:** Change all permutations of `batch system`, `batch_sys` etc to an equivalent `job runner`, apart from where `batch system` refers to the _workflow_ config item `[runtime][X][job]batch system`.

I've tried to _batch_ the changes into somewhat digestible commits for reviewing.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).

- [x] Appropriate tests are included (functional).

- [ ] Appropriate change log entry included.

- [ ] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.

- [x] No dependency changes.

### CODE REVIEW

File: cylc/flow/batch_sys_handlers/loadleveler.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """Submits task job scripts to loadleveler by the ``llsubmit`` command.
 
+# TODO: rewrite the following for platforms:
+
```
Comment (MetRonnie) [inline cylc/flow/batch_sys_handlers/loadleveler.py:19]: Presumably these code block examples need to be updated from the `[runtime][X][job]` to `[platforms]`?

File: cylc/flow/data_messages.proto [hunk 2]
```diff
@@ -108,8 +108,8 @@ message PbJob {
     optional string submitted_time = 6;
     optional string started_time = 7;
     optional string finished_time = 8;
-    optional string batch_sys_job_id = 9;
-    optional string batch_sys_name = 10;
+    optional string job_id = 9;
+    optional string job_runner_name = 10;
```
Comment (MetRonnie) [inline cylc/flow/data_messages.proto:112]: @dwsutherland Can I get a check from you that this protobuf stuff has been done correctly?

File: cylc/flow/batch_sys_handlers/loadleveler.py (hunk 1 above)
Comment (oliver-sanders) [reply cylc/flow/batch_sys_handlers/loadleveler.py:19]: Yep, https://github.com/cylc/cylc-flow/issues/3997

File: tests/functional/directives/README [hunk 3]
```diff
@@ -1,4 +1,6 @@
 
+# TODO: Does this need to be removed?
```
Comment (oliver-sanders) [inline tests/functional/directives/README:2]: Yeah, kill it.

**Comment by oliver-sanders** [comment]
Looks good, very thorough!

**Comment by hjoliver** [comment]
The failed test `tests/f/cylc-poll/04-poll-multi-hosts.t` was fixed by #3995 (so a rebase will get it).

File: cylc/flow/data_messages.proto (hunk 2 above)
Comment (dwsutherland) [reply cylc/flow/data_messages.proto:112]: Looks good to me, as long as the instructions at the top were followed, the binary used to generate the module is the same version protobuf, and tests pass 👍

Comment (dwsutherland) [reply cylc/flow/data_messages.proto:112]: The diff in the generated module looks appropriate.. Will re-generate on review and see if there are any changes 👍

**Comment by oliver-sanders** [comment]
One small hitch with Cylc env vars, after that should be good to go.

File: tests/functional/job-file-trap/00-sigusr1.t [hunk 4]
```diff
@@ -34,11 +34,11 @@ run_tests() {
     # Make sure t1.1.1's status file is in place
     T1_STATUS_FILE="${SUITE_RUN_DIR}/log/job/1/t1/01/job.status"
 
-    poll_grep -E 'CYLC_BATCH_SYS_JOB_ID=' "${T1_STATUS_FILE}"
+    poll_grep -E 'CYLC_JOB_ID=' "${T1_STATUS_FILE}"
```
[the remark this answers is not in the document]
Comment (MetRonnie) [reply tests/functional/job-file-trap/00-sigusr1.t:37]: Are these actually environment variables? I can see they get written to `job.status` but can't find any indication of such env vars being set

[the remark this answers is not in the document]
Comment (hjoliver) [reply tests/functional/job-file-trap/00-sigusr1.t:37]: The `job.status` file can be read and `eval`'d line by line to define environment variables - but I don't think we actually do that anywhere, do we @oliver-sanders ?

**Review by hjoliver — APPROVED** [review]
Looks good :+1:

File: tests/functional/job-file-trap/00-sigusr1.t (hunk 4 above)
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply tests/functional/job-file-trap/00-sigusr1.t:37]: Ah sorry, put this comment in a misleading place. I thought we exposed `CYLC_BATCH_SYS_JOB_ID` to the job execution environment, however, it looks like we are safe.

**Review by oliver-sanders — APPROVED** [review]

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**